src/DynamoCore/Extensions/LinterExtensionBase.cs

```diff
@@ -245,7 +246,7 @@ private void UnlinkFromCurrentWorkspace()
             if (currentWorkspace != null)
             {
                 UnsubscribeGraphEvents(currentWorkspace);
-                linterManager.RuleEvaluationResults.Clear();
+                DynamoModel.OnRequestDispatcherInvoke(() => { this.linterManager.RuleEvaluationResults.Clear(); });
```

Comment:
I believe these are not needed and we only need the dispatcherInvoke for the changes in the binded collection in the view model.